File: BackEnd/java/ZaloMini/src/main/java/com/devmam/zalomini/entities/Notification.java
package com.devmam.zalomini.entities;

import jakarta.persistence.*;
import jakarta.persistence.Table;
import lombok.*;
import org.hibernate.annotations.*;
import org.hibernate.type.SqlTypes;

import java.time.Instant;
import java.util.Date;
import java.util.Map;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity(name = "Notification")
@Table(name = "notifications")
public class Notification {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "type", nullable = false, length = 50)
    private String type;

    @Column(name = "title", nullable = false)
    private String title;

    @Lob
    @Column(name = "content")
    private String content;

    @Column(name = "data")
    @JdbcTypeCode(SqlTypes.JSON)
    private Map<String, Object> data;

    @ColumnDefault("0")
    @Column(name = "is_read")
    private Boolean isRead;

    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "created_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;

}

File: BackEnd/Zalo_mini/Zalo_mini/Models/notification.cs
using System;
using System.Collections.Generic;

namespace Zalo_mini.Models
{
    public partial class notification
    {
        public long id { get; set; }
        public long user_id { get; set; }
        public string type { get; set; } = null!;
        public string title { get; set; } = null!;
        public string? content { get; set; }
        /// <summary>
        /// Additional data as JSON
        /// </summary>
        public string? data { get; set; }
        public bool? is_read { get; set; }
        public DateTime? created_at { get; set; }

        public virtual user user { get; set; } = null!;
    }
}
